src/NuGet.Core/NuGet.LibraryModel/LibraryDependency.cs

```diff
@@ -38,5 +42,37 @@ public bool HasFlag(LibraryDependencyTypeFlag flag)
         {
             return Type.Contains(flag);
         }
+
+        public static void AddLibraryDependency(LibraryDependency dependency, ISet<LibraryDependency> list)
+        {
```

Comment:
@rohit21agrawal this method should be moved to a pack helper, it should not be part of the librarymodel since this is specific to pack and makes assumptions based on pack.